tikv-client/src/main/java/com/pingcap/tikv/txn/Lock.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.pingcap.tikv.txn;
+
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.kvproto.Kvrpcpb;
+
+public class Lock {
+  private long txnID;
+  private long ttl;
+  private ByteString key;
+  private ByteString primary;
+  private static final long defaultLockTTL = 3000;
+
+  public Lock(Kvrpcpb.LockInfo l) {
+    txnID = l.getLockVersion();
+    key = l.getKey();
+    primary = l.getPrimaryLock();
+    ttl = l.getLockTtl() == 0 ? defaultLockTTL : l.getLockTtl();
+  }
+
+  public long getTxnID() {
+    return txnID;
+  }
+
+  public void setTxnID(long txnID) {
```

Comment:
I searched around, there is almost no need to change the status of Lock once created. If so, make this an immutable object. Usually immutable is preferred since you don't need to bother hashcode change, reasoning field change, multi-threading and etc.